packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -807,9 +807,6 @@ class _RestorableAutovalidateMode extends RestorableValue<AutovalidateMode> {
 /// * Single Date picker with calendar mode.
 /// * Single Date picker with text input mode.
 /// * Date Range picker with text input mode.
-///
-/// [helpText], [orientation], [icon], [onIconPressed] are required and must be
```

Comment:
Most of these are no longer relevant. And this should be obvious from the signature.